File: 0532-k-diff-pairs-in-an-array/0532-k-diff-pairs-in-an-array.cpp
class Solution {
public:
    int findPairs(vector<int>& nums, int k) {
        int res = 0;
        std::unordered_set<int> seen_sums;
        
        for (size_t i = 0; i < nums.size(); ++i) {
            for (size_t j = i + 1; j < nums.size(); ++j) {
                if (std::abs(nums[i] - nums[j]) == k) {
                    int sum = nums[i] + nums[j];
                    if (seen_sums.find(sum) == seen_sums.end()) {
                        seen_sums.insert(sum);
                        res++;
                    }
                }
            }
        }
        
        return res;
    }
};

File: 0532-k-diff-pairs-in-an-array/0532-k-diff-pairs-in-an-array.py
class Solution:
    def findPairs(self, nums: List[int], k: int) -> int:
        count=0
        seen=[]
        for i in range(len(nums)):
            for j in range(i+1,len(nums)):
                if abs(nums[i]-nums[j])==k and [nums[i]+nums[j]] not in seen:
                    seen.append([nums[i]+nums[j]])
                    count+=1
             
        return count